src/Halcyon/HAL/HALResponse.cs

```diff
@@ -26,9 +25,9 @@ public class HALResponse {
         public HALResponse(object model, IHALModelConfig config = null)
             : this(config) {
             if (!(model is JObject) && (model is IEnumerable)) {
-                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", "model");
+                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", nameof(model));
             }
-
+            HALAttributeResolver.ResolveAttributes(this, model);
```

Comment:
However I need to modify the HALResponse so that the private field dto is accessible to the IHALConverter. For now I've made the field Internal ReadOnly which I hope is ok with you.